src/components/crownBoundaryEventDropdown.vue

```diff
@@ -24,6 +25,9 @@
           @click="addBoundaryEvent(nodeType)"
```

Comment:
I wonder if it would be cleaner to add the invalid text to `dropdownData`, and re-use the logic for the `:disabled` binding to display this text. This would eliminate the need for the added conditional logic in this component.

So the only addition to this file would be changinig the `b-tooltip` component to:
```html
<b-tooltip
  v-if="!canAddBoundaryEventToTarget(dropdownData.nodeType, shape)"
  :target="nodeType"
  variant="warning"
  placement="right"
>
  {{ dropdownData.disabledLabel }}
</b-tooltip>
```

**Note:** `dropdownData` is destructured in the `v-for` on the parent `li` but I referenced it directly in my example above.

This would work well with the changes just merged in https://github.com/ProcessMaker/modeler/pull/953, as the text would only have to be added to the objects in `boundaryEventDropdownData` in `task.vue`.

Reply:
@chipit24 I think this is a great solution and is better than reaching for polymorphism right now. :)

Reply:
Only issue is, I was trying not to display the "Can only add one per Task" for Error Message, when the real warning is "No available space"

Reply:
I pushed up changes based on this recommendation. Hopefully they show up. Github is acting weird.